File: 01_Language/01_Functions/js/array_pop.js
function array_pop(arr) {
    if (Array.isArray(arr)) {
        return arr.pop()
    }

    let lastKey;
    for (let i in arr) {
        lastKey = i
    }
    if (lastKey === undefined) {
        return null;
    }

    let res = arr[lastKey]
    delete arr[lastKey]
    return res
}

let stack = ["orange", "banana", "apple", "raspberry"]
console.log(array_pop(stack), stack)

let map = {"k1": "orange", "k2": "banana", "k3": "apple", "k4": "raspberry"}
console.log(array_pop(map), map)

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/ArrayPop.java
package com.onevgo.functions;

import java.util.*;

public class ArrayPop {
    public static <E> E arrayPop(List<E> list) {
        if (list.size() == 0) {
            return null;
        }

        return list.remove(list.size() - 1);
    }

    public static <E> E arrayPop(Stack<E> stack) {
        return stack.size() > 0 ? stack.pop() : null;
    }

    public static <K, V> V arrayPop(LinkedHashMap<K, V> map) {
        if (map.size() == 0) {
            return null;
        }

        K lastKey = null;
        for (K k : map.keySet()) {
            lastKey = k;
        }
        return map.remove(lastKey);
    }

    public static void main(String[] args) {
        List<String> stack = new ArrayList<>(Arrays.asList("orange", "banana", "apple", "raspberry"));
        System.out.println(arrayPop(stack));
        System.out.println(stack);

        LinkedHashMap<Object, Object> map = new LinkedHashMap<>();
        map.put("k1", "orange");
        map.put("k2", "banana");
        map.put("k3", "apple");
        map.put("k4", "raspberry");
        System.out.println(arrayPop(map));
        System.out.println(map);
    }
}
